src/ActiveLogin.Authentication.BankId.AspNetCore/Areas/BankIdAuthentication/Controllers/BankIdApiController.cs

```diff
@@ -361,13 +359,13 @@ public async Task<ActionResult> Cancel(BankIdLoginApiCancelRequest request)
                 await _bankIdApiClient.CancelAsync(orderRef.OrderRef);
                 await _bankIdEventTrigger.TriggerAsync(new BankIdCancelSuccessEvent(orderRef.OrderRef));
             }
-            catch (Exception exception)
+            catch (BankIdApiException exception)
             {
                 // When we get exception in a cancellation request, chances
                 // are that the orderRef has already been cancelled or we have
                 // a network issue. We still want to provide the GUI with the
                 // validated cancellation URL though.
-                await _bankIdEventTrigger.TriggerAsync(new BankIdCancelFailedEvent(orderRef.OrderRef, exception));
+                await _bankIdEventTrigger.TriggerAsync(new BankIdCancelErrorEvent(orderRef.OrderRef, exception));
             }
 
```

Comment:
Should catch all other Exceptions?
```suggestion
catch (Exception exception) 
    {
        // Log or something
     }
```